gstd/src/msg/encoded.rs

```diff
@@ -25,14 +25,41 @@ use crate::{
     errors::{Error, IntoResult, Result},
     msg::{utils, CodecMessageFuture, MessageFuture},
     prelude::{
-        convert::AsRef,
         mem::{transmute, MaybeUninit},
         ops::RangeBounds,
     },
     ActorId, MessageId, ReservationId,
 };
 use gstd_codegen::wait_for_reply;
-use scale_info::scale::{Decode, Encode, MaxEncodedLen, Output};
+use scale_info::scale::{Decode, Encode, Output};
+
+fn with_optimized_encode<T, E: Encode>(payload: E, f: impl FnOnce(&[u8]) -> T) -> T {
+    struct ExternalBufferOutput<'a> {
+        buffer: &'a mut [MaybeUninit<u8>],
+        offset: usize,
+    }
+
+    impl<'a> Output for ExternalBufferOutput<'a> {
+        fn write(&mut self, bytes: &[u8]) {
+            const ERROR_LOG: &str = "Unexpected encoding behavior: too large input bytes size";
+            let end_offset = self.offset.checked_add(bytes.len()).expect(ERROR_LOG);
+            if end_offset > self.buffer.len() {
+                panic!("{ERROR_LOG}");
```

Comment:
Add str right here, what's the point to format anything?

Reply:
It was written by Gregory, I just made change:
```diff
- panic!("{}", ERROR_LOG);
+ panic!("{ERROR_LOG}");
```
We can't use just string literal here:
```
error: format argument must be a string literal
 --> src/main.rs:3:12
  |
3 |     panic!(ERROR_LOG);
  |            ^^^^^^^^^
```

Reply:
So why do you need this const? Add some simple expl and copy-paste

Reply:
Maybe move this conversation to #2975?

Reply:
kk

Reply:
> Maybe move this conversation to #2975?

will we merge 2975 first?

Reply:
Yes, this is an additional PR to 2975, because we solved the problem with increasing the size of the binary file.